### PR TITLE
Remove PrivateCredentials class, move functionality into APIKeys class

### PR DESCRIPTION
**Pull request by evangriffiths**
Fixes https://github.com/gnosis/prediction-market-agent-tooling/issues/218

### CODE REVIEW

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 1]
```diff
@@ -206,19 +203,17 @@
 
     def redeem_positions(
         self,
-        private_credentials: PrivateCredentials,
+        api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:205]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 2]
```diff
@@ -370,7 +365,7 @@
 
 
 def omen_buy_outcome_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:367]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 3]
```diff
@@ -423,15 +418,15 @@
 
 
 def binary_omen_buy_outcome_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:420]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 4]
```diff
@@ -441,7 +436,7 @@
 
 
 def omen_sell_outcome_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:438]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 5]
```diff
@@ -482,36 +477,34 @@
...
         # Optionally, withdraw from the collateral token back to the `from_address` wallet.
-        collateral_token.withdraw(
-            private_credentials=private_credentials, amount_wei=amount_wei, web3=web3
-        )
+        collateral_token.withdraw(api_keys=api_keys, amount_wei=amount_wei, web3=web3)
 
 
 def binary_omen_sell_outcome_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:498]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 6]
```diff
@@ -627,13 +618,13 @@
 
 
 def omen_fund_market_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:620]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 7]
```diff
@@ -644,24 +635,24 @@
...
+    market_contract.addFunding(api_keys, funds, web3=web3)
 
 
 def build_parent_collection_id() -> HexStr:
     return HASH_ZERO  # Taken from Olas
 
 
 def omen_redeem_full_position_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:654]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 8]
```diff
@@ -741,7 +732,7 @@
 
 
 def omen_remove_fund_market_tx(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:734]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 9]
```diff
@@ -802,13 +791,13 @@
 
 
 def redeem_from_all_user_positions(
-    private_credentials: PrivateCredentials,
+    api_keys: APIKeys,
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:793]: I disagree

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 10]
```diff
@@ -136,10 +136,8 @@ def place_bet(
         if amount.currency != self.currency:
             raise ValueError(f"Omen bets are made in xDai. Got {amount.currency}.")
         amount_xdai = xDai(amount.amount)
-        keys = APIKeys()
-        private_credentials = PrivateCredentials.from_api_keys(keys)
         binary_omen_buy_outcome_tx(
-            private_credentials=private_credentials,
+            api_keys=APIKeys(),
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen.py:140]: Whaaat? It's necessary!

**Review by kongzii — APPROVED** [review]

**Pull request merged**